app/code-area/page.tsx

```diff
@@ -25,69 +26,86 @@ const CodeArea = () => {
     );
     const router = useRouter();
     const { mosaicThemeDark } = useCodeAreaStore();
+    const { email } = useStepperStore();
+    const { check } = useStepperStore();
+    const { progress } = useStepperStore();
+    const { challenge } = useStepperStore();
 
-    return (
-        <div className='h-screen'>
-            <div className='pointer-events-auto absolute inset-0 z-20 bg-white bg-opacity-50 xl:hidden' />
-            <div className='absolute inset-0 z-30 flex items-center justify-center xl:hidden'>
-                <div className='w-3/4 max-w-lg rounded-xl bg-black p-6 shadow-lg'>
-                    <h2 className='mb-4 text-center text-xl font-bold text-white'>
-                        It looks like you&quot;re on a small screen
-                    </h2>
-                    <p className='text-sm font-semibold text-gray-400'>
-                        We hate to be non-inclusive towards phones and tablets,
-                        however we want to provide you with the best experience
-                        possible!
-                    </p>
-                    <div className='flex justify-center'>
-                        <Button
-                            variant='destructive'
-                            className='mt-2'
-                            onClick={() => router.push("/")}
-                        >
-                            Back to main page...
-                        </Button>
+    if (
+        email === "" &&
+        check === false &&
+        progress !== (99 || 100) &&
+        challenge === ""
+    ) {
+        return (
+            <p>
+                error, no bypassing the required form to enter the coding area!
```

Comment:
Needless to say, UI needs a boost. But more importantly, provide a method for users to go back to wherever they came from originally. They don't need to necessarily come from the home page but for the time being just navigating back to home will do. Maybe even a countdown timer that redirects them back to home would be cool too?

Reply:
done